src/map/roe.cpp

```diff
@@ -317,11 +353,131 @@ bool SetEminenceRecordProgress(CCharEntity* PChar, uint16 recordID, uint32 progr
+
+        {   // Daily Reset
+            auto* jst = gmtime(&jstnow);
+            jst->tm_hour = 0;
+            jst->tm_min = 0;
+            jst->tm_sec = 0;
+            auto lastJstMidnight = timegm(jst) - JST_OFFSET;     // Unix timestamp of the last JST midnight
+
+            if (lastOnline < lastJstMidnight)
+                ClearDailyRecords(PChar);
+        }
+
+        {   // 4hr Reset
+            auto* jst = gmtime(&jstnow);
+            jst->tm_hour = jst->tm_hour & 0xFC;
+            jst->tm_min = 0;
+            jst->tm_sec = 0;
+            auto lastJstTimedBlock = timegm(jst) - JST_OFFSET;   // Unix timestamp of the start of the current 4-hr block
+
+            if (lastOnline < lastJstTimedBlock || PChar->m_eminenceLog.active[30] != GetActiveTimedRecord())
+                AddActiveTimedRecord(PChar);
+        }
+    }
+}
+
+uint16 GetActiveTimedRecord()
+{
+    uint8 day = static_cast<uint8>(CVanaTime::getInstance()->getJstWeekDay());
+    uint8 block = static_cast<uint8>(CVanaTime::getInstance()->getJstHour() / 4);
+    return RoeSystem.TimedRecordTable[day][block];
+}
+
+void AddActiveTimedRecord(CCharEntity* PChar)
+{
+    // Clear old timed entries from log
+    PChar->m_eminenceLog.progress[30] = 0;
+    PChar->m_eminenceCache.activemap &= ~RoeSystem.TimedRecords;
+
+    // Add current timed record to slot 31
+    auto timedRecordID = GetActiveTimedRecord();
+    PChar->m_eminenceLog.active[30] = timedRecordID;
+    PChar->m_eminenceCache.activemap.set(timedRecordID);
+    PChar->pushPacket(new CRoeUpdatePacket(PChar));
+
+    SetEminenceRecordCompletion(PChar, timedRecordID, false);
+}
+
+void ClearDailyRecords(CCharEntity* PChar)
+{
+    // Set daily record progress to 0
+    for (int i = 0; i < 30; i++)
+    {
+        if (auto recordID = PChar->m_eminenceLog.active[i]; RoeSystem.DailyRecords.test(recordID))
+        {
+            PChar->m_eminenceLog.progress[i] = 0;
+        }
+    }
+    PChar->pushPacket(new CRoeUpdatePacket(PChar));
+
+    // Set completion for daily records to 0
+    for (auto record : RoeSystem.DailyRecordIDs)
+    {
+        uint16 page = record / 8;
+        uint8 bit = record % 8;
+        PChar->m_eminenceLog.complete[page] &= ~(1 << bit);
+    }
+
+    charutils::SaveEminenceData(PChar);
+
+    for (int i = 0; i < 4; i++)
+        PChar->pushPacket(new CRoeQuestLogPacket(PChar, i));
+}
+
+void CycleTimedRecords()
+{
+    if (!RoeSystem.RoeEnabled)
+        return;
+
+    zoneutils::ForEachZone([](CZone* PZone){
+        PZone->ForEachChar([](CCharEntity* PChar){
+            if (GetEminenceRecordCompletion(PChar, 1))
+            {
+                AddActiveTimedRecord(PChar);
+            }
+        });
+    });
+}
+
+void CycleDailyRecords()
+{
+    if (!RoeSystem.RoeEnabled)
+        return;
+
+    zoneutils::ForEachZone([](CZone* PZone){
+        PZone->ForEachChar([](CCharEntity* PChar){
+            ClearDailyRecords(PChar);
+        });
+    });
+}
```

Comment:
`BreakBeforeBraces: Allman` & `Lambdas`

Reply:
Feel free to ignore trying to style these, we're still trying to find a style that doesn't fly in the face of all the other rules, that is actually reproducible by tooling (Lamdas & formatting tooling are not buddies)